sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala

```diff
@@ -309,7 +309,8 @@ class DataFrameReader private[sql](sparkSession: SparkSession) extends Logging {
     val parts: Array[Partition] = predicates.zipWithIndex.map { case (part, i) =>
       JDBCPartition(part, i) : Partition
     }
-    val relation = JDBCRelation(parts, options)(sparkSession)
+    val schema = JDBCRelation.getSchema(sparkSession.sessionState.conf.resolver, options)
+    val relation = JDBCRelation(schema, parts, options)(sparkSession)
```

Comment:
We do not need to change this. Add an apply function to object JDBCRelation.scala
```
  def apply(parts: Array[Partition], jdbcOptions: JDBCOptions)(
      sparkSession: SparkSession): JDBCRelation = {
    val schema = getSchema(jdbcOptions, sparkSession.sessionState.conf.resolver)
    JDBCRelation(schema, parts, jdbcOptions)(sparkSession)
  }
```

Reply:
ok